clicommand/pipeline_upload.go

```diff
@@ -236,6 +241,23 @@ var PipelineUploadCommand = cli.Command{
 			return
 		}
 
+		if len(cfg.RedactedVars) > 0 {
+			needles := redaction.GetValuesToRedact(shell.StderrLogger, cfg.RedactedVars, env.FromSlice(os.Environ()).ToMap())
+			serialisedPipeline, err := result.MarshalJSON()
+
+			if err != nil {
+				l.Fatal("Pipeline serialization of \"%s\" failed (%s)", src, err)
```

Comment:
What do you think about tweaking this error to say something like:

`"Couldn’t scan the %q pipeline for redacted variables. This parsed pipeline could not be serialized, ensure the pipeline YAML is valid, or disable secret redaction for this upload by passing --redacted-vars=''. (%s)", src, err`

Reply:
Yea, I like that. It's hitting the three points of what makes a good error message. I'll update! :)